server/app/controllers/admin/AdminApiKeysController.java

```diff
@@ -0,0 +1,77 @@
+package controllers.admin;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.Authorizers;
+import auth.CiviFormProfile;
+import auth.ProfileUtils;
+import controllers.CiviFormController;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.pac4j.play.java.Secure;
+import play.data.DynamicForm;
+import play.data.FormFactory;
+import play.mvc.Http;
+import play.mvc.Result;
+import services.apikey.ApiKeyCreationResult;
+import services.apikey.ApiKeyService;
+import services.program.ProgramService;
+import views.admin.apikeys.ApiKeyNewOneView;
+import views.admin.programs.ApiKeyCredentialsView;
+
+/** Controller for admins managing ApiKeys. */
+public class AdminApiKeysController extends CiviFormController {
+
+  private final ApiKeyService apiKeyService;
+  private final ApiKeyNewOneView newOneView;
+  private final ApiKeyCredentialsView apiKeyCredentialsView;
+  private final ProgramService programService;
+  private final FormFactory formFactory;
+  private final ProfileUtils profileUtils;
+
+  @Inject
+  public AdminApiKeysController(
+      ApiKeyService apiKeyService,
+      ApiKeyNewOneView newOneView,
+      ApiKeyCredentialsView apiKeyCredentialsView,
+      ProgramService programService,
+      FormFactory formFactory,
+      ProfileUtils profileUtils) {
+    this.apiKeyService = checkNotNull(apiKeyService);
+    this.newOneView = checkNotNull(newOneView);
+    this.apiKeyCredentialsView = checkNotNull(apiKeyCredentialsView);
+    this.programService = checkNotNull(programService);
+    this.formFactory = checkNotNull(formFactory);
+    this.profileUtils = checkNotNull(profileUtils);
+  }
+
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result index() {
+    return ok();
+  }
+
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
```

Comment:
Not-blocking: Out of curiosity, if all handlers in a controller require the same authorizer, is it possible to annotate at the class level instead? This would help prevent the case where a new handler is added to an existing controller and forgot to add an authorizer. It would potentially also reduce code duplication.

Reply:
The docs don't seem to suggest that you can: https://github.com/pac4j/play-pac4j/wiki/Apply-security

I tried moving the annotation to the class and while the app did run, I got some strange behavior when trying to access the secured routes as an authorized user.

Reply:
No worries. Thanks for digging.